File: siserp-logistica-backend/src/main/java/com/besoft/siserp/lgtc/dto/request/CompraModificarRequest.java
package com.besoft.siserp.lgtc.dto.request;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class CompraModificarRequest implements Serializable {
	private static final long serialVersionUID = 1L;

	private Long id;
	private Long idProveedor;
	private Long idtTipoComprobante;
	private Double montoTotal;
	private String serieComprobante;
	private String nroComprobante;
	private String nroOrdenCompra;
	private String observacion;
	private Date fecha;
	private Integer flagActivo;
	private Long idUsuarioMod;
	private Date fecUsuarioMod;
	private List<CompraModificarDetalleRequest> listaDetalleCompra;

	private String nomProveedor;
	private Long tipDocProveedor;
	private String nroDocProveedor;

}

File: siserp-logistica-frontend/src/main/siserp-logistica-frontend/src/app/modules/intranet/dto/request/compra-modificar-request.ts
import { CompraModificarDetalleRequest } from "./compra-modificar-detalle.request";
import { CompraRegistrarDetalleRequest } from "./compra-registrar-detalle.request";

export class CompraModificarRequest {
    id: number;
    idProveedor: number;
    idtTipoComprobante: number;
    montoTotal: number;
    serieComprobante: string;
    nroComprobante: string;
    nroOrdenCompra: string;
    observacion: string;
    fecha: Date;
    flagActivo: number;
    idUsuarioMod: number;
    fecUsuarioMod: Date;
    listaDetalleCompra: CompraModificarDetalleRequest[];

    nomProveedor: string;
    tipDocProveedor: number;
    nroDocProveedor: string;

}